dev/PushNotifications/PushNotificationRegistrationToken.cpp

```diff
@@ -0,0 +1,24 @@
+﻿#include "pch.h"
+#include "PushNotificationRegistrationToken.h"
+#include "Microsoft.Windows.PushNotifications.PushNotificationRegistrationToken.g.cpp"
+
+#include <Windows.ApplicationModel.background.h>
+
+namespace winrt::Microsoft::Windows::PushNotifications::implementation
+{
+    PushNotificationRegistrationToken::PushNotificationRegistrationToken(uint64_t cookie, winrt::Windows::ApplicationModel::Background::BackgroundTaskRegistration const& taskRegistration)
+    {
+        m_cookie = cookie;
+        m_taskRegistration = taskRegistration;
+    }
+    uint64_t PushNotificationRegistrationToken::Cookie()
+    {
+        auto lock = m_lock.lock_shared();
```

Comment:
No need for a lock once you make the members `const` and construct them rather than assign them in the constructor.